Lib/multiprocessing/pool.py

```diff
@@ -145,6 +147,21 @@ def _helper_reraises_exception(ex):
 # Class representing a process pool
 #
 
+class _PoolCache(dict):
+    """
+    Class that implements a cache for the Pool class that will notify
+    the pool management threads every time the cache is emptied. The
+    notification is done by the use of a queue that is provided when
+    instantiating the cache.
+    """
+    def __init__(self, *args, notifier=None, **kwds):
+        self.notifier = notifier
+        super().__init__(*args, **kwds)
+    def __delitem__(self, item):
+        super().__delitem__(item)
+        if not self:
+            self.notifier.put(None)
```

Comment:
Can you add a comment explaining why it's important to wake up when the cache is emptied?